crates/router/src/workflows/payment_method_mandate_details_revoke.rs

```diff
@@ -0,0 +1,169 @@
+            "PaymentMethodMandateRevokeTrackingData",
+        )?;
+
+        let key_store = db
+            .get_merchant_key_store_by_merchant_id(
+                &tracking_data.merchant_id,
+                &db.get_master_key().to_vec().into(),
+            )
+            .await?;
+
+        let retry_count = process.retry_count;
+        let merchant_account = db
+            .find_merchant_account_by_merchant_id(&tracking_data.merchant_id, &key_store)
+            .await?;
+
+        let connector_name = tracking_data.connector.to_string();
+        let merchant_connector_account = helpers::get_merchant_connector_account(
+            state,
+            &tracking_data.merchant_id,
+            None,
+            &key_store,
+            &tracking_data.profile_id,
+            &connector_name,
+            Some(&tracking_data.merchant_connector_id),
+        )
+        .await?;
+        let connector_data = ConnectorData::get_connector_by_name(
+            &state.conf.connectors,
+            &connector_name,
+            GetToken::Connector,
+            Some(tracking_data.merchant_connector_id.clone()),
+        )?;
+        let connector_integration: services::BoxedMandateRevokeConnectorIntegrationInterface<
+            types::api::MandateRevoke,
+            types::MandateRevokeRequestData,
+            types::MandateRevokeResponseData,
+        > = connector_data.connector.get_connector_integration();
+
+        let router_data = utils::construct_mandate_revoke_router_data(
+            merchant_connector_account,
+            &merchant_account,
+            tracking_data.customer_id.clone(),
+            &connector_name,
+            Some(tracking_data.connector_mandate_id),
+            None,
+        )
+        .await?;
+
+        let response = services::execute_connector_processing_step(
+            state,
+            connector_integration,
+            &router_data,
+            CallConnectorAction::Trigger,
+            None,
+        )
+        .await
+        .change_context(errors::ApiErrorResponse::InternalServerError)?;
+        match response.response {
+            Ok(_mandate) => {
+                metrics::CONNECTOR_MANDATE_REVOKE_SUCCEEDED.add(
+                    &metrics::CONTEXT,
+                    1,
+                    &add_attributes([("flow", "")]),
+                );
+                db.as_scheduler()
+                    .finish_process_with_business_status(process, business_status::COMPLETED_BY_PT)
+                    .await?;
+            }
+            Err(err) => {
+                metrics::CONNECTOR_MANDATE_REVOKE_FAILED.add(
+                    &metrics::CONTEXT,
+                    1,
+                    &add_attributes([(
+                        "flow",
+                        format!(
+                            "connector_name = {:?}, error_code = {:?}",
+                            &connector_name, err.code
+                        ),
+                    )]),
```

Comment:
Same here, this can be two different attributes, `connector` and `error_code`, with corresponding values.